tests/lib/rules/arrow-parens.js

```diff
@@ -133,6 +156,44 @@ var invalid = [
             message: asNeededMessage,
             type: type
         }]
+    },
+
+    // "as-needed", { "requireForBlockBody": true }
+    {
+        code: "a => {}",
+        output: "(a) => {}",
+        options: ["as-needed", {requireForBlockBody: true}],
+        parserOptions: { ecmaVersion: 6 },
+        errors: [{
+            line: 1,
+            column: 1,
+            message: requireForBlockBodyNoParensMessage,
+            type: type
+        }]
+    },
+    {
+        code: "(a) => a",
+        output: "a => a",
+        options: ["as-needed", {requireForBlockBody: true}],
+        parserOptions: { ecmaVersion: 6 },
+        errors: [{
+            line: 1,
+            column: 1,
+            message: requireForBlockBodyMessage,
+            type: type
+        }]
+    },
+    {
+        code: "(b) => b",
+        output: "b => b",
+        options: ["as-needed", {requireForBlockBody: true}],
+        parserOptions: { ecmaVersion: 6 },
+        errors: [{
+            line: 1,
+            column: 1,
+            message: requireForBlockBodyMessage,
+            type: type
+        }]
     }
```

Comment:
Could we add a test that this is invalid: `(a) => ({});`. Don't mean to harp on that one case, but it seems like one that is more likely to break if someone ever touches this in the future. Would also love to see a test for destructured params in the invalid tests!